src/store/blockstore-level.ts

```diff
@@ -40,20 +40,24 @@ export class BlockstoreLevel implements Blockstore {
     return new BlockstoreLevel({ ...this.config, location: '' }, db);
   }
 
-  async put(key: CID, val: Uint8Array, options?: Options): Promise<void> {
-    return this.db.put(key.toString(), val, options);
+  async isEmpty(options?: Options): Promise<boolean> {
+    return this.db.isEmpty(options);
   }
 
-  async get(key: CID, options?: Options): Promise<Uint8Array> {
-    return this.db.get(key.toString(), options);
+  async put(key: CID | string, val: Uint8Array, options?: Options): Promise<void> {
+    return this.db.put(String(key), val, options);
   }
 
-  async has(key: CID, options?: Options): Promise<boolean> {
-    return !! await this.get(key, options);
+  async get(key: CID | string, options?: Options): Promise<Uint8Array> {
```

Comment:
for my understanding: I don't understand how adding `| string` would work (though it clearly does), the type declaration for `Blockstore` says:

`export interface Blockstore extends Store<CID, Uint8Array>`

so I though the key has to be a `CID`? Is the type definition out of date, or am I understanding things incorrectly?

Reply:
im pretty sure the `|` is an "or", so it still satisfies the requirement from `Blockstore` that the `key` is a `CID` while also allowing for `string` (which ultimately is what it gets turned into anyways inside `BlockstoreLevel` for use with `LevelWrapper`)